### PR TITLE
feature: support FIRST to mark the beginning of test

### PR DESCRIPTION
**Pull request by spacewander**
FIRST is the contrary of LAST.

### CODE REVIEW

File: inc/Test/Base.pm [hunk 1]
```diff
@@ -438,14 +438,32 @@ sub _block_list_init {
 
 sub _choose_blocks {
     my $blocks = [];
+    my $has_first = 0;
+
+    for my $hunk (@_) {
+        my $block = $self->_make_block($hunk);
+        if (exists $block->{FIRST}) {
+            $has_first = 1;
+            last;
```
[the remark this answers is not in the document]
Comment (spacewander) [reply inc/Test/Base.pm:447]: This loop here only detects if has_first is true. It is not relative to the queue.

**Review by doujiang24 — APPROVED** [review]

File: inc/Test/Base.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhuizhuhaomeng) [reply inc/Test/Base.pm:447]: .The next loop does not include the FIRST marked test case, so the FIRST marked test case was not included in the final test cases.

It is so confusing.

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: inc/Test/Base.pm [hunk 2]
```diff
@@ -438,14 +438,32 @@ sub _block_list_init {
...
                 unless $self->_no_diag_on_only;
             return [$block];
         }
+
         next if exists $block->{SKIP};
+        if ($has_first) {
+            if (exists $block->{FIRST}) {
+                $has_first = 0;
+            }
+            next;
```
Comment (doujiang24) [inline inc/Test/Base.pm:464]:
```suggestion
            } else {
                next;
            }
```
Okay, I got you @zhuizhuhaomeng 
Do you mean this change?

Comment (zhuizhuhaomeng) [reply inc/Test/Base.pm:464]: Yes

Comment (zhuizhuhaomeng) [reply inc/Test/Base.pm:464]: Yes, that's what I mean

**Review by zhuizhuhaomeng — APPROVED** [review]

**Pull request merged**